app/jobs/scheduled_publishing_job.rb

```diff
@@ -9,32 +9,37 @@ class ScheduledPublishingJob < ApplicationJob
   discard_and_log(ActiveRecord::RecordNotFound)
 
   def perform(id)
-    published_edition = Edition.find_and_lock_current(id: id) do |edition|
-      return if no_longer_schedulable?(edition) # rubocop:disable Lint/NonLocalExitFromIterator
+    edition = nil
+
+    Edition.transaction do
+      edition = Edition.lock.find_current(id: id)
+      return unless publishable?(edition)
 
       user = edition.status.created_by
       reviewed = edition.status.details.reviewed
       PublishService.new(edition)
                     .publish(user: user, with_review: reviewed)
     end
 
-    send_notifications(published_edition)
+    send_notifications(edition)
   end
 
 private
 
-  def no_longer_schedulable?(edition)
+  def publishable?(edition)
```

Comment:
Minor, but the new name is a little bit misleading, since the edition is 'publishable'. I was wondering whether it would be helpful to rename this to something like 'cancelled?', since we rely on it for that?